File: MATH375-NumericalAnalysis/HW8/trapezoidal.m
function integral = trapezoidal(f, a, b, n)
    h = (b-a)/n;
    x = linspace (a, b, n+1);
    eval = zeros(1, n);

    for i = 1:n+1
        eval(i) = feval(f, x(i));
    end

    w = [0 ones(1,n)]+[ones(1,n) 0];

    integral = (h/2)*sum(w.*eval);
end

File: CS471-ScientificComputing/hw3/code/trapezoidal.py
from scipy import *
from numpy import *
from scipy import integrate


def trapezoidal(f, lower, upper, n):
    h = (upper - lower) / n
    x = linspace(lower, upper, n + 1)
    feval = zeros(n+1)

    for i in range(0, n+1):
        feval[i] = f(x[i])
    
    weights = ones(n+1)
    for i in range(1, n):
        weights[i] += 1
    
    integral = (h / 2) * sum( multiply(weights, feval) )

    return integral
